### ISSUE TITLE
'spot search' does not work for me

### ISSUE THREAD

**Issue by ephbaum** [phpfunk/alfred-spotify-controls#29]
I would normally try to fix this type of issue and push it back up but I haven't the time. So, I'll open an issue in the hopes that someone else can track this down. 

I am running php 5.4.24 on my system, which is a 13" MBPr running OS X 10.9.4. 

Here's the problem: 

In Alfred I type spot search and add a space and then when I begin typing my query Alfred flips over to my default web search solutions. 
# tl;dr w/ pictures:
### Everything seems to be looking good here:

![looking_good](https://cloud.githubusercontent.com/assets/4037628/3528874/9f213fc0-0793-11e4-85bb-41043c43bc97.png)
### But then it's all gone wrong:

![all_gone_wrong](https://cloud.githubusercontent.com/assets/4037628/3528882/b4166464-0793-11e4-9b59-64a3a48969c3.png)

Thanks a lot!

**Comment by phpfunk** [comment]
If you look at the hint below the first screenshot you have to type `spot search [album|artist|track] QUERY` so for this it would be: `spot search artist weezer`. The reason for this is because Spotify makes you search all three different APIs for results.

You could search all three but it would take 3x the amount of time to get results. Also if you use the search keyboard shortcut (you can set this up in the workflow (mine is command+enter), it will bring up a Spotify branded Alfred box where you would just have to type: `[album|artist|track] QUERY`. So in this case it would be `artist weezer`

This is also outlined in the wiki: https://github.com/phpfunk/alfred-spotify-controls/wiki/Searching

**Comment by ephbaum** [comment]
I'll take a look at this in a bit, though a quick attempt now (having tried this before...) still has the same results. Perhaps I just need to bind a key to get it to work as expected. Thanks for the reply!

![google_will_find_weezer_for_me](https://cloud.githubusercontent.com/assets/4037628/3530353/9d43d676-07a3-11e4-80dd-c08dcaff7ec3.png)

**Comment by phpfunk** [comment]
Sometimes it just takes a bit I think. I just tried here and got this.
![screen shot 2014-07-09 at 4 06 36 pm](https://cloud.githubusercontent.com/assets/72203/3530456/99382144-07a4-11e4-81f8-980fb578fa85.png)

**Comment by ephbaum** [comment]
Ah, well... I think this is probably what's wrong (from the debug console in Alfred): 

![the_brokens](https://cloud.githubusercontent.com/assets/4037628/3530483/ef538514-07a4-11e4-9be4-cb96780e3fcf.png)

**Comment by phpfunk** [comment]
ACK! I didn't see you were using images. I will try that and take a look at the above. Thanks for posting that.

---

Verified with images is the issue. I will fix this up. I would default to NO for now until I fix that.

**Comment by phpfunk** [comment]
@fskirschbaum I pushed a fix for images here: https://github.com/phpfunk/alfred-spotify-controls/releases/tag/2.2.9

I still don't use them, far too slow but it is fixed so it should work as intended now.

**Comment by ephbaum** [comment]
I guess I'm not following... I'm using images, what does that mean in this context?

**Comment by phpfunk** [comment]
Sorry, search worked for me w/o using images. I then saw you were searching with images and tried and it was broke.

I fixed search with images but just note it will be slower because it has to download the images from spotify to show in the results.

If you download 2.2.9, search with images is fixed.

**Comment by ephbaum** [comment]
![still_so_confused](https://cloud.githubusercontent.com/assets/4037628/3530790/3ac5e34a-07a8-11e4-97d7-2e57c0c185c2.png)

I'm still a bit confused, I suppose. I'm still having the same issues it seems... perhaps I've completely missed a step somehow.

**Comment by ephbaum** [comment]
And, to be clear... I have tried it through the hotkey functionality as well: 

![sanity_check](https://cloud.githubusercontent.com/assets/4037628/3530825/bd8eec18-07a8-11e4-926e-b726d2f346d2.png)

**Comment by phpfunk** [comment]
Do you have any type of services/urls being blocked on the network you are on? It seems you are not getting any results from the API.

This is a completely different error than I saw because you aren't making it to the download images portion of the script where I saw the error.

Can you get results here: http://ws.spotify.com/search/1/artist.json?q=weezer

Also what version of PHP are you on my chance? you need curl and json installed, which they should be by default.

**Comment by ephbaum** [comment]
I can run Spotify just fine, but it's entirely possible something is being blocked. However, I tried it out on hurl.it and it seems to be working fine using your url structure from search.php: 

![such_response](https://cloud.githubusercontent.com/assets/4037628/3530898/9929d2b0-07a9-11e4-950a-7d93edd0c9f3.png)

**Comment by phpfunk** [comment]
Just try it in your browser, it should render. For some reason you are getting no results returned. I am going to fix that error you see, but you will still get no results.

**Comment by ephbaum** [comment]
Sorry I didn't make it clear... it does work in my browser: 

![browser](https://cloud.githubusercontent.com/assets/4037628/3530952/2a2e723e-07aa-11e4-9633-cf8489d8f64f.png)

**Comment by ephbaum** [comment]
And I get the same from a curl: 

![curly](https://cloud.githubusercontent.com/assets/4037628/3530982/7be409ae-07aa-11e4-8883-fd26d75af672.png)

**Comment by phpfunk** [comment]
ok cool. we can rule that out. Can  you try this out:

Open the spotify workflow, double click the spot search box, then click 'Open workflow folder' button. Open the search.php file and on line 21 or so place this:

```
print $json;
exit;
```

save the file then search with the bug window in alfred open. Let me know what you see.

Sorry I just don't have the same issue. hard to debug.

**Comment by ephbaum** [comment]
Ah, cool. Well. I'm ashamed to admit it, but it was my company's blocker... I'm not sure why it's blocked within this script and not in browser or some other method. Thank you for your help. 

![websense_of_doom](https://cloud.githubusercontent.com/assets/4037628/3531144/5c833cae-07ac-11e4-81bc-25b0da446a24.png)

**Comment by ephbaum** [comment]
Found it. My company is blocking open.spotify.com and when trying to get Artwork you have to use this getTrackArtwork method which of course fails.

**Comment by phpfunk** [comment]
@fskirschbaum hey I'm just glad we found the issue. I am going to actually make a fix for this if it can't get JSON for either place it will tell the user. So in the end this is a good find :) Thx for your help here.

**Comment by ephbaum** [comment]
No problem. I see the issue is that I get a response from websense and it looks like in your php you are only checking for an empty response to test for error in reply. 

Keen. Great workflow! Thank you for your help!

**Comment by phpfunk** [comment]
You are 100% correct :) and I am glad you enjoy it.
